clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -231,25 +246,23 @@ public NetworkClientDelegate.UnsentRequest toUnsentRequest() {
 
     private class OffsetFetchRequestState extends RequestState {
         public final Set<TopicPartition> requestedPartitions;
-        public final GroupState.Generation requestedGeneration;
-        private final CompletableFuture<Map<TopicPartition, OffsetAndMetadata>> future;
+        public CompletableFuture<Map<TopicPartition, OffsetAndMetadata>> future;
+
         public OffsetFetchRequestState(final Set<TopicPartition> partitions,
-                                       final GroupState.Generation generation,
```

Comment:
We do need member epoch (and member ID), to be included in the request if we have them, exactly as it is done for the OffsetCommit.

Reply:
My comment is really only to make sure we are on the same page regarding the need of epoch here. But ok if we don't include the changes in this PR, because I also have them in the changes for OffsetFetch v9 that will follow this. I added id and epoch here not only to send it in the request, but also to differentiate the requests that will be retried.

Reply:
For the current impl, it sends a generation of -1 and id of "".  I see both fields are "ignorable", not sure if that means we don't need to send them, @dajac - I assume ignorable means we don't need to send them if we don't have one?

Reply:
Yes, agree we don't need to send them if we don't have them (member not using the group management capabilities), but the point is member ID/epoch should be here to be included if the member has them.